src/pytkdocs/loader.py

```diff
@@ -913,3 +957,39 @@ def field_is_inherited(field_name: str, fields_name: str, base_class: type) -> b
             *(getattr(parent_class, fields_name, {}).keys() for parent_class in base_class.__mro__[1:-1]),
         ),
     )
+
+
+def split_attr_name(attr_name: str) -> Tuple[str, Optional[str]]:
+    """
+    Split an attribute name into a first-order attribute name and remainder.
+
+    Args:
+        attr_name: Attribute name (a)
+
+    Returns:
+        Tuple containing:
+            first_order_attr_name: Name of the first order attribute (a)
+            remainder: The remainder (b.c)
+
+    """
+    first_order_attr_name, *remaining = attr_name.split(".", maxsplit=1)
+    remainder = remaining[0] if remaining else None
+    return first_order_attr_name, remainder
+
+
+def get_fields(attr_name: str, *, members: Mapping[str, Any] = None, class_obj=None) -> Dict[str, Any]:
+    if not (bool(members) ^ bool(class_obj)):
+        raise ValueError("Either members or class_obj is required.")
+    first_order_attr_name, remainder = split_attr_name(attr_name)
+    fields = members[first_order_attr_name] if members else dict(vars(class_obj)).get(first_order_attr_name, {})
+    if remainder:
+        fields = attrgetter(remainder)(fields)
+
+    if callable(fields):
+        fields = fields()
+
+    if not isinstance(fields, dict):
+        #  Support Django models
+        fields = {getattr(f, "name", str(f)): f for f in fields}
```

Comment:
That's because Django's `_meta.fields` is an `ImmutableList`, it is possible to retrieve all the fields using `model.__dict__`, but Django applies some magic to them and I figured that it is much easier to retrieve them from `_meta.fields`.

```python
>>> from api.models import StopGroup
>>> StopGroup.__dict__
mappingproxy({'__module__': 'api.models', '__doc__': '\n    Grupa przystanków ZTM.\n    ', 'manager_attr': 'stop_groups', 'from_schedule': <classmethod object at 0x7fc23fa65490>, '__str__': <function StopGroup.__str__ at 0x7fc23faacf70>, '_meta': <Options for StopGroup>, 'DoesNotExist': <class 'api.models.StopGroup.DoesNotExist'>, 'MultipleObjectsReturned': <class 'api.models.StopGroup.MultipleObjectsReturned'>, 'id': <django.db.models.query_utils.DeferredAttribute object at 0x7fc23fa65580>, 'name': <django.db.models.query_utils.DeferredAttribute object at 0x7fc23fa65610>, 'region_id': <django.db.models.query_utils.DeferredAttribute object at 0x7fc23fa65670>, 'region_name': <django.db.models.query_utils.DeferredAttribute object at 0x7fc23fa656d0>, 'objects': <django.db.models.manager.ManagerDescriptor object at 0x7fc23fa65850>, 'stop_set': <django.db.models.fields.related_descriptors.ReverseManyToOneDescriptor object at 0x7fc23fa65940>})
>>> StopGroup._meta.fields
(<django.db.models.fields.IntegerField: id>, <django.db.models.fields.CharField: name>, <django.db.models.fields.CharField: region_id>, <django.db.models.fields.CharField: region_name>)
>>> StopGroup._meta.get_fields()
(<ManyToOneRel: api.stop>, <django.db.models.fields.IntegerField: id>, <django.db.models.fields.CharField: name>, <django.db.models.fields.CharField: region_id>, <django.db.models.fields.CharField: region_name>)
```

Reply:
I've just realized that `_meta.fields` doesn't include `ManyToMany` fields. I think that filtering `get_fields` with `not field.auto_created` would be a better approach.

Unfortunately, the is not much in the docs about `_meta` API. 
In the docstring for `_meta.fields`, they state that it is a "private API intended only to be used by Django itself" and "get_fields() combined with filtering of field properties is the public API for this field list."

By the way, it's interesting that `_meta.concrete_fields` has a very similar disclaimer and yet is included in an example here:  https://docs.djangoproject.com/en/3.2/ref/models/instances/#django.db.models.Model.from_db.

Reply:
Thank you for this detailed explanation :slightly_smiling_face: